frontend/src/metabase/dashboard/containers/DashboardSharingEmbeddingModal.jsx

```diff
@@ -49,36 +44,20 @@ class DashboardSharingEmbeddingModal extends Component {
       isLinkEnabled,
       updateEnableEmbedding,
       updateEmbeddingParams,
+      onClose,
       ...props
     } = this.props;
     if (!enabled) {
       return null;
     }
     return (
-      <ModalWithTrigger
-        ref={m => (this._modal = m)}
+      <Modal
         full
         disabled={!isLinkEnabled}
-        as={ModalTrigger}
-        triggerElement={
-          <span
-            className={linkClassNames}
-            aria-disabled={!isLinkEnabled}
-            onClick={() => {
-              if (isLinkEnabled) {
-                MetabaseAnalytics.trackStructEvent(
-                  "Sharing / Embedding",
-                  "dashboard",
-                  "Sharing Link Clicked",
-                );
-              }
-            }}
-          >
-            {linkText}
-          </span>
-        }
         triggerClasses={className}
         className="scroll-y"
+        as={ModalTrigger}
```

Comment:
I checked that this component eventually renders https://github.com/metabase/metabase/blob/master/frontend/src/metabase/components/Modal/FullPageModal.tsx and I don't see `as` being sent anywhere.

I think this should be removable.

```suggestion
```

That also means `DashboardSharingEmbeddingModal.styled` should be removable.